packages/headless/src/features/folding/folding-slice.ts

```diff
@@ -86,28 +86,31 @@ function resolveRootFromFields(
 
 function createCollectionFromResult(
   relevantResult: ResultWithFolding,
-  fields: FoldingFields
+  fields: FoldingFields,
+  rootResult?: ResultWithFolding
 ): FoldedCollection {
   const resultsInCollection = getAllIncludedResultsFrom(relevantResult);
 
-  const rootResult =
+  const resolvedRoot =
     resolveRootFromFields(resultsInCollection, fields) ?? relevantResult;
```

Comment:
I just realized now that it's not necessary to call `resolveRootFromFields` when `rootResult` is specified.

All in all, the code could look something like this:
```ts
const resultsInCollection = getAllIncludedResultsFrom(relevantResult);

const resultToUseAsRoot = rootResult
  ?? resolveRootFromFields(resultsInCollection, fields)
  ?? relevantResult;

return {
  result: resultToUseAsRoot,
  children: resolveChildrenFromFields(
    resultToUseAsRoot,
    resultsInCollection,
    fields
  ),
  moreResultsAvailable: true,
  isLoadingMoreResults: false
};
```